### PR TITLE
Backend table update, validation and tests

### PR DESCRIPTION
**Pull request by jesushilden**

### CODE REVIEW

**Comment by jesushilden** [comment]
Korjaan traviksen...

**Comment by myzozoz** [comment]
controllers/highscore.js:43:
typo `intallationIdTaken` -> `installationIdTaken`
sama myös rivillä 47

File: controllers/highscores.js [hunk 1]
```diff
@@ -31,4 +34,33 @@ highscoreRouter.post('/', async (request, response) => {
     }
 })
 
+const validate = async (highscore) => {
+    const errorMessages = []
+
+    const validationError = highscore.validateSync()
+    const usernameTaken = await Highscore.findOne({ user: highscore.user })
+    const tokenTaken = await Highscore.findOne({ token: highscore.token })
+    const intallationIdTaken = await Highscore.findOne({ installationId: highscore.installationId })
```
Comment (myzozoz) [inline controllers/highscores.js]:
```suggestion
    const installationIdTaken = await Highscore.findOne({ installationId: highscore.installationId })
```

File: controllers/highscores.js [hunk 2]
```diff
@@ -31,4 +34,33 @@ highscoreRouter.post('/', async (request, response) => {
...
+
+    if (usernameTaken) errorMessages.push('Username already taken')
+    if (tokenTaken) errorMessages.push('Token already taken')
+    if (intallationIdTaken) errorMessages.push('InstallationId already taken')
```
Comment (myzozoz) [inline controllers/highscores.js]:
```suggestion
    if (installationIdTaken) errorMessages.push('InstallationId already taken')
```

**Review by myzozoz — APPROVED** [review]
Code changes reviewed.

**Pull request merged**